### PR TITLE
Truncating non future estimates

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?
Forcing all project update estimates to be in the future or "0001-01-01T00:00:00Z" (no estimate). During a project update, estimates of 1970 and a few minutes in the past are provided when the process is started. This change removes these bad estimates. 

### :chains: Related Resources
#703

### :+1: Definition of Done
No estimates in the past are returned. 

### :athletic_shoe: How to Build and Test the Change
1. `build components/authz-service && start_all_services`
1. Add compliance reports with `chef_load_compliance_scans -N1000 -D11 -M5 -T22000`
1. Turn on projects `chef-automate iam upgrade-to-v2 --beta2.1`
1. Create a project and then update all the projects. To watch the percentage and estimated time complete run `chef-automate dev grpcurl authz-service -- chef.automate.domain.authz.v2.Projects.ApplyRulesStatus`
I like to use `watch "chef-automate dev grpcurl authz-service -- chef.automate.domain.authz.v2.Projects.ApplyRulesStatus"` But you have to install watch first with `install_if_missing core/procps-ng watch`

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: components/authz-service/server/v2/project_update_manager.go [hunk 1]
```diff
@@ -432,3 +404,45 @@ func (m *CerealProjectUpdateManager) Status() (ProjectUpdateStatus, error) {
 	}
 	return projectUpdateInstance, nil
 }
+
+// Find the longest estimate that is in the future. If no estimate is found return zero time.Time{}
+func findLongestEstimatedTime(estimatedTimes []time.Time) time.Time {
+	longestEstimatedTimeComplete := time.Time{}
+	for _, estimatedTime := range estimatedTimes {
+		if estimatedTime.After(time.Now()) && estimatedTime.After(longestEstimatedTimeComplete) {
```
Comment (srenatus) [inline components/authz-service/server/v2/project_update_manager.go]: It's probably inconsequential, but `time.Now()` will be different for every iteration through this loop. Maybe we should define `noew := time.Now()` before that loop and compare to this here? 💭

File: components/authz-service/server/v2/project_update_manager.go [hunk 2]
```diff
@@ -432,3 +404,45 @@ func (m *CerealProjectUpdateManager) Status() (ProjectUpdateStatus, error) {
...
+
+func (w *workflowInstance) collectAllDomainServiceEstimates() []time.Time {
+	estimates := make([]time.Time, 0)
+	if !w.IsRunning() {
+		return estimates
+	}
+	domainServicesUpdateInstance, err := w.GetUpdateDomainServicesInstance()
+	if err == cereal.ErrWorkflowInstanceNotFound {
+		return estimates
+	}
```
Comment (srenatus) [inline components/authz-service/server/v2/project_update_manager.go]: ❓ Can there be other errors that we might care about? -- Can `err` be not `nil` and not `cereal.ErrWorkflowInstanceNotFound`?

File: components/authz-service/server/v2/project_update_manager.go [hunk 3]
```diff
@@ -432,3 +404,45 @@ func (m *CerealProjectUpdateManager) Status() (ProjectUpdateStatus, error) {
...
+			logrus.WithError(err).Errorf("failed to get subworkflow for %q", d)
+			continue
+		}
+		payload := project_update_tags.DomainProjectUpdateWorkflowPayload{}
+		if subWorkflow.IsRunning() {
+			if err := subWorkflow.GetPayload(&payload); err != nil {
+				logrus.WithError(err).Errorf("failed to get payload for %q", d)
+				continue
+			}
+			estimatedTime := time.Unix(payload.MergedJobStatus.EstimatedEndTimeInSec, 0)
```
Comment (srenatus) [inline components/authz-service/server/v2/project_update_manager.go]: For some reason I don't know about, we know that `payload.MergedJobStatus` is not a thing that could be `nil` and `EstimatedEndTimeInSec` is always available?

File: components/authz-service/server/v2/project_update_manager.go (hunk 2 above)
Comment (lancewf) [reply components/authz-service/server/v2/project_update_manager.go]: I am not sure. Maybe @jaym might know this question.

Comment (jaym) [reply components/authz-service/server/v2/project_update_manager.go]: i think not. but it doesn't hurt to defend against it

File: components/authz-service/server/v2/project_update_manager.go (hunk 3 above)
Comment (lancewf) [reply components/authz-service/server/v2/project_update_manager.go]: That makes sense. I will add a check to see if it is nil.

Comment (lancewf) [reply components/authz-service/server/v2/project_update_manager.go]: Wait,`payload.MergedJobStatus` can not be nil, but `payload.MergedJobStatus.EstimatedEndTimeInSec` could be zero.

**Pull request merged**